ckanext/protected_resources/plugin.py

```diff
@@ -142,8 +115,15 @@ def protected_resource_unlock(context, data_dict):
 
     resource = p.toolkit.get_action('resource_show')(context, {'id': data_dict['resource_id']})
 
-    if 'is_protected' in resource:
-        del resource['is_protected']
+    if not resource_has_protected_status(data_dict['resource_id']):
+        raise p.toolkit.ValidationError(['This resource is already unlocked'])
+    else:
+        log.error("WE're on the right track my dude")
+        qry = "DELETE FROM resource_protected WHERE resource_id = :resource_id"
+        model.Session.execute(qry, {
+            'resource_id': data_dict['resource_id']
+        })
+        model.Session.commit()
```

Comment:
What happens if this query throws an error, say for example the `resource_id` didn't exist in the `resource_protected` table?

Reply:
A 500 error will be thrown, and a raygun notification will arrive with us, I think this is acceptable. Thoughts?